caddyhttp/httpserver/replacer.go

```diff
@@ -232,15 +232,23 @@ func (r *replacer) getSubstitution(key string) string {
 	case "{path}":
 		// if a rewrite has happened, the original URI should be used as the path
 		// rather than the rewritten URI
-		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
-		if path == "" {
+		var path string
+		origpath := r.request.Header.Get("Caddy-Rewrite-Original-URI")
+		if origpath == "" {
 			path = r.request.URL.Path
+		} else {
+			origURL, _ := url.Parse(origpath)
```

Comment:
I am assuming that it is safe to ignore err here as Caddy-Rewrite-Origional-URI should always be a valid URL.  Is this ok?